src/main/java/com/yahoo/bullet/rest/common/Reader.java

```diff
@@ -4,48 +4,49 @@
  *  See the LICENSE file associated with the project for terms.
  */
 
-package com.yahoo.bullet.rest.query;
+package com.yahoo.bullet.rest.common;
 
-import com.yahoo.bullet.pubsub.Metadata;
 import com.yahoo.bullet.pubsub.PubSubMessage;
+import com.yahoo.bullet.pubsub.PubSubResponder;
 import com.yahoo.bullet.pubsub.Subscriber;
 import lombok.extern.slf4j.Slf4j;
 
-import java.util.Arrays;
-import java.util.HashSet;
-import java.util.Set;
-import java.util.concurrent.ConcurrentMap;
+import java.util.Objects;
 
 @Slf4j
-public class PubSubReader {
+public class Reader {
     private Subscriber subscriber;
-    private ConcurrentMap<String, QueryHandler> requestQueue;
-    private Thread readerThread;
+    private PubSubResponder responder;
+    private Thread reader;
     private int sleepTimeMS;
-    private static final Set<Metadata.Signal> FINISHED =
-            new HashSet<>(Arrays.asList(Metadata.Signal.KILL, Metadata.Signal.COMPLETE, Metadata.Signal.FAIL));
-    // TODO: Handle Subscribers that have failed and we have no more readers
-
     /**
      * Create a service with a {@link Subscriber} and a request queue.
      *
      * @param subscriber The Subscriber to read responses from.
-     * @param requestQueue The {@link ConcurrentMap} containing open requests.
+     * @param responder The {@link PubSubResponder} to use to respond.
```

Comment:
It's misleading making this a PubSubResponder because it's actually always QueryService which contains the list of PubSubResponders used...

Reply:
Well this class doesn't know that. It's just coding to the interface. I didn't intend PubSubResponder to only be used as a thing that responds to async results from the backend. I meant it as a thing that knows what to do with a PubSubMessage in general.